lib/noticed/base.rb

```diff
@@ -111,13 +111,28 @@ def get_class(name, options)
       end
     end
 
-    # Validates that all params are present
     def validate!
+      validate_params_present!
+      validate_email_delivery_methods_have_mailers_present!
```

Comment:
Architecture question... should we loop through each delivery method and let them define validations?

I don't feel like the Notification base class is the right place to validate the mailer options, you know? Seems like it would make more sense to live in the `DeliveryMethods::Email` class. 

We could pass in the options to a class method on the delivery method for validation?

```
module DeliveryMethods
  class Base
    def self.validate!(options)
      # Override to raise ValidationError if passed options are invalid
    end
  end
end
```

class Noticed::Base
  def validate!
    delivery_methods.each do |method|
       # convert to class and call klass.validate!(method[:options])
    end
  end
end

Reply:
Related https://github.com/excid3/noticed/pull/34

Reply:
I ❤️  this idea, let me rework on this 🙂